File: src/main/java/com/shunya/reverse/service/MathService.java
package com.shunya.reverse.service;

import org.springframework.stereotype.Service;

@Service
public class MathService {
    
    public long reverse(long num) {
        long sum = 0;
        while (num != 0) {
            long lastDigit = (num % 10);
            sum = (sum * 10) + lastDigit;
            num = num / 10;
        }
        return sum;
    }
}

File: src/main/kotlin/com/shunya/reverse/MathService.kt
package com.shunya.reverse

import org.springframework.stereotype.Service

@Service
class MathService {

    fun reverse(num: Long): Long {
        var num = num
        var sum: Long = 0
        while (num != 0L) {
            val lastDigit = num % 10
            sum = sum * 10 + lastDigit
            num /= 10
        }
        return sum
    }
}
